src/components/select/select-list/select-list.component.js

```diff
@@ -258,30 +258,32 @@ const SelectList = React.forwardRef(
 
     const handleGlobalKeydown = useCallback(
       (event) => {
-        const { key } = event;
-        const isActionButtonFocused =
-          document.activeElement === listActionButtonRef.current;
-
-        if (key === "Tab" && listActionButton) {
-          handleActionButtonTab(event, isActionButtonFocused);
-        } else if (key === "Tab" || key === "Escape") {
-          onSelectListClose();
-        } else if (key === "Enter" && !isActionButtonFocused) {
-          event.preventDefault();
-          const currentOption = childrenList[currentOptionsListIndex];
+        if (isOpen) {
```

Comment:
instead of wrapping the whole block you could just not handle that when the element is not open:
```
if (!isOpen) {
  return;
}
```